controllers/vault.go

```diff
@@ -0,0 +1,281 @@
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/go-errors/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	utilexec "k8s.io/client-go/util/exec"
+)
+
+//
+// {
+// 	"unseal_keys_b64": [
+// 	  "R/BL306DUjRQIHdkYYxheqFxr6PtZVEKtHaYNjFqBGq7",
+// 	  "+4CYavmqRWq165WJM4DqpnEqlDnECt6q+6jSmEaJBsBA",
+// 	  "tlsQ833l5k52ESK28jlZlWbegBRY+HNIJD9Yqp3cEdF6",
+// 	  "ON0wQUleo+iW4r6U0EwmoOkRhezzTke09h+rxgRDPkdo",
+// 	  "i1hmENQhAcq5t6WWxTR35YDAUjY1w8ry751CggPsB0Jk"
+// 	],
+// 	"unseal_keys_hex": [
+// 	  "47f04bdf4e83523450207764618c617aa171afa3ed65510ab4769836316a046abb",
+// 	  "fb80986af9aa456ab5eb95893380eaa6712a9439c40adeaafba8d298468906c040",
+// 	  "b65b10f37de5e64e761122b6f239599566de801458f87348243f58aa9ddc11d17a",
+// 	  "38dd3041495ea3e896e2be94d04c26a0e91185ecf34e47b4f61fabc604433e4768",
+// 	  "8b586610d42101cab9b7a596c53477e580c0523635c3caf2ef9d428203ec074264"
+// 	],
+// 	"unseal_shares": 5,
+// 	"unseal_threshold": 3,
+// 	"recovery_keys_b64": [],
+// 	"recovery_keys_hex": [],
+// 	"recovery_keys_shares": 5,
+// 	"recovery_keys_threshold": 3,
+// 	"root_token": "s.lAR1G890NPBEkzRt8Ic5kBVz"
+//   }
+// Note: We only keep the minimum needed fields around
+type VaultInitStruct struct {
+	UnsealKeysHex   []string `json:"unseal_keys_hex"`
+	UnsealShares    int      `json:"unseal_shares"`
+	UnsealThreshold int      `json:"unseal_threshold"`
+	RootToken       string   `json:"root_token"`
+}
+
+// {
+// 	"type": "shamir",
+// 	"initialized": false,
+// 	"sealed": true,
+// 	"t": 0,
+// 	"n": 0,
+// 	"progress": 0,
+// 	"nonce": "",
+// 	"version": "1.9.2",
+// 	"migration": false,
+// 	"recovery_seal": false,
+// 	"storage_type": "file",
+// 	"ha_enabled": false,
+// 	"active_time": "0001-01-01T00:00:00Z"
+//   }
+type VaultStatus struct {
+	Type        string `json:"type"`
+	Initialized bool   `json:"initialized"`
+	Sealed      bool   `json:"sealed"`
+	T           int    `json:"t"`
+	N           int    `json:"n"`
+	Progress    int    `json:"progress"`
+	Version     string `json:"version"`
+	StorageType string `json:"storage_type"`
+	HaEnabled   bool   `json:"ha_enabled"`
+}
+
+const vaultSecretName string = "vaultkeys"
+const operatorNamespace string = "patterns-operator-system"
+
+func vaultOperatorInit(config *rest.Config, client kubernetes.Interface) (*VaultInitStruct, error) {
+	stdout, stderr, err := execInPod(config, client, "vault", "vault-0", "vault", []string{"vault", "operator", "init", "-format=json"})
+	if err != nil {
+		newerr := fmt.Errorf("%v - %s - %s", err, stdout, stderr)
+		return nil, newerr
+	}
+	var unmarshalled VaultInitStruct
+	err = json.Unmarshal(stdout.Bytes(), &unmarshalled)
+	if err != nil {
+		newerr := fmt.Errorf("%v - %s - %s", err, stdout, stderr)
+		return nil, newerr
+	}
+	return &unmarshalled, nil
+}
+
+func addVaultSecrets(config *rest.Config, client kubernetes.Interface, vaultInitOutput *VaultInitStruct) error {
```

Comment:
```suggestion
func createVaultSecrets(config *rest.Config, client kubernetes.Interface, vaultInitOutput *VaultInitStruct) error {
```

`add` had me thinking you were adding keys into vault

Reply:
aye, makes sense